### PR TITLE
udev: Add partition symlinks

### PR DESCRIPTION
**Pull request by jepio**
The current rule only creates a symlink for each device. Extend it to create symlinks for partitions too.

### CODE REVIEW

**Comment by jepio** [comment]
Outcome looks something like this:
```
# find /dev/disk/azure/
/dev/disk/azure/
/dev/disk/azure/os
/dev/disk/azure/os-part4
/dev/disk/azure/os-part9
/dev/disk/azure/os-part3
/dev/disk/azure/os-part2
/dev/disk/azure/os-part6
/dev/disk/azure/os-part7
/dev/disk/azure/os-part1
/dev/disk/azure/local
/dev/disk/azure/local/by-name
/dev/disk/azure/local/by-name/nvme-440G-2
/dev/disk/azure/local/by-name/nvme-440G-1
/dev/disk/azure/local/by-name/nvme-440G-1-part1
/dev/disk/azure/local/by-index
/dev/disk/azure/local/by-index/2
/dev/disk/azure/local/by-index/1
/dev/disk/azure/local/by-index/1-part1
```

File: udev/80-azure-nvme.rules.in [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ACTION!="add|change", GOTO="azure_nvme_end"
 SUBSYSTEM!="block", GOTO="azure_nvme_end"
 KERNEL!="nvme*", GOTO="azure_nvme_end"
-ATTR{nsid}!="?*", GOTO="azure_nvme_end"
+ENV{ID_NSID}!="?*", GOTO="azure_nvme_end"
```
Comment (cjp256) [inline udev/80-azure-nvme.rules.in:4]: out of curiosity is there a reason when to use one method over the other?

**Review by cjp256 — COMMENTED** [review]
awesome, thanks!

File: udev/80-azure-nvme.rules.in (hunk 1 above)
Comment (jepio) [reply udev/80-azure-nvme.rules.in:4]: `ATTR{nsid}` is only present on the disk (nvme0n1), so it doesn't match for partitions. You can either use `ATTRS{nsid}` to get matching against parent attributes, or switch to `ENV{ID_NSID}`. Udev itself ships the rule that does:
```
ATTRS{nsid}=="?*",         ENV{ID_NSID}="$attr{nsid}"
``` 
so I'd think that they want people using `ENV{ID_NSID}`. The same applies to `ENV{ID_MODEL}`, it's synthesized from `ATTRS{model}`.

Comment (cjp256) [reply udev/80-azure-nvme.rules.in:4]: thanks! makes sense, I figured there was a good reason :D

**Review by cjp256 — APPROVED** [review]
Works great!

$ find /dev/disk/azure/ | sort
/dev/disk/azure/
/dev/disk/azure/local
/dev/disk/azure/local/by-index
/dev/disk/azure/local/by-index/1
/dev/disk/azure/local/by-index/1-part1
/dev/disk/azure/local/by-index/2
/dev/disk/azure/local/by-name
/dev/disk/azure/local/by-name/nvme-440G-1
/dev/disk/azure/local/by-name/nvme-440G-1-part1
/dev/disk/azure/local/by-name/nvme-440G-2
/dev/disk/azure/os
/dev/disk/azure/os-part1
/dev/disk/azure/os-part14
/dev/disk/azure/os-part15

**Pull request merged**